### PR TITLE
ARXIVNG-2881: add error message about large file uploads

### PR DESCRIPTION
**Pull request by mhl10**
This PR:

- upgrades arxiv-base to v0.16.5, both to bring submission up to date and to incorporate Erick's changes to support [custom errors in the default backend](https://arxiv-org.atlassian.net/browse/ARXIVNG-2585). This upgrade may have created some unintentional styling and layout problems (see screenshot below)
- catches exceptions related to large file uploads. When running the UI in the Flask development environment, uploading a large file usually causes a "connection reset" exception to be thrown. In a production environment (e.g. behind nginx), the exception will normally be `RequestEntityTooLarge`.

Screenshot 2020-01-27 14 21 25

### CODE REVIEW

File: submit/controllers/ui/new/upload.py [hunk 1]
```diff
@@ -113,8 +119,25 @@ def upload_files(method: str, params: MultiDict, session: Session,
...
         # Otherwise, treat this as an upload attempt.
         logger.debug('POST; user is uploading file(s)')
-        return _post_upload(params, files, session, submission, rdata, token)
-    raise MethodNotAllowed('Nope')
+        try:
+            return _post_upload(params, files, session,
+                                submission, rdata, token)
+        except exceptions.ConnectionFailed as ex:
+            logger.debug('Problem POSTing upload: %s', ex)
+            alerts.flash_failure(Markup(
```
Comment (mhl10) [inline submit/controllers/ui/new/upload.py:127]: This is this the message one will typically see in the Flask dev environment.

File: submit/controllers/ui/new/upload.py [hunk 2]
```diff
@@ -113,8 +119,25 @@ def upload_files(method: str, params: MultiDict, session: Session,
...
+                'There was a problem uploading your file. '
+                f'{PLEASE_CONTACT_SUPPORT}'
+            ))
+            return _get_upload(params, session, submission, rdata, token)
+
+        except RequestEntityTooLarge as ex:
+            logger.debug('Problem POSTing upload: %s', ex)
+            alerts.flash_failure(Markup(
+                'There was a problem uploading your file because it exceeds '
```
Comment (mhl10) [inline submit/controllers/ui/new/upload.py:136]: This is the message one should see in the production environment.

**Comment by SBBCornell** [comment]
Quick review based on the screenshot: it does look like some messages meant for screen readers only are displaying. I'll review more deeply...

**Review by SBBCornell — COMMENTED** [review]
So far I found this visual error: the style "sr-only" is being used in some places but doesnt exist in the css. It's causing some elements for screen readers to display for everyone. The style that should be used is called "is-sr-only". Each style is in use in different places. I can work on finding and replacing tomorrow morning.

**Review by SBBCornell — APPROVED** [review]
Testing a large file upload returns an error now, not a 500 page, so appears to be working! 

Ive also reviewed further for unintended visual changes and the only error I found was related to the "sr-only" class. I'll make that small fix in a new branch based off of develop so it doesnt conflict with the latest layout changes.

**Review by bdc34 — APPROVED** [review]
The code looks good. I didn't test with a large upload.

**Comment by DavidLFielding** [comment]
Code looks fine. I was originally going to skip actually testing this but decided to take time for a quick test.

My problem was with base. After fixing the UI catches large files without ugly 500 error. Looks good!

Side note: Changing base seems a little tedious. I'm not sure if there is a way to simplify this. Should we be doing this system-wide?

**Comment by DavidLFielding** [comment]
I'm running against the latest develop branch which contains the file upload error improvements from this PR.

I'm only seeing the red error box at the top of the page when I upload a massive file. I'm not seeing red warning box down below or any details about the actual error. Am I supposed to see something about the uploaded file being too large?

I'm not getting a 500 error, which is good, but I have no idea why my upload failed. Well, I know, but the system is not reminding me.[Correction: I was just able to generate a 500 error while uploading a large file but I'm not sure what happened yet. Looks like some sort of read timeout where I see timeout value set to 30 seconds.]

arxiv-submission-ui       |     raise ReadTimeoutError(self, url, "Read timed out. (read timeout=%s)" % timeout_value)
arxiv-submission-ui       | urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='arxiv-filemanager', port=8000): Read timed out. (read timeout=30)

Hmm. I'm also able to upload several large files for a total of 60MB without errors. I believe we relaxed the lower legacy oversize limits. This seems off. I will need to double check.

**Comment by mhl10** [comment]
> I'm not getting a 500 error, which is good, but I have no idea why my upload failed. Well, I know, but the system is not reminding me

It looks like you're running in to the case where the `ConnectionResetError` is being caught, which is expected. I wasn't sure it was safe to assume that is this _always_ caused by a large file--hence the generic error message. In the production environment we should be catching the `RequestEntityTooLarge` HTTP error, which is accompanied by a more specific error message.

**Pull request merged**